core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/value/nearby/NearValueNearbyValueSelector.java

```diff
@@ -103,11 +103,19 @@ public Iterator<Object> iterator() {
         }
     }
 
+    public Iterator<Object> fixedNearbyOriginIterator(Object origin) {
+        if (!randomSelection) {
+            return new FixedOriginOriginalNearbyValueIterator(nearbyDistanceMatrixSupply, childSelector.getSize(), origin);
+        } else {
+            return new FixedOriginRandomNearbyValueIterator(nearbyDistanceMatrixSupply, nearbyRandom, workingRandom, origin,
+                    childSelector.getSize());
+        }
```

Comment:
In another PR (https://github.com/kiegroup/optaplanner/pull/2752) we're actually enabling the replaying selector to be able to call the original twice.

I'm wondering - should we use the same approach here as we use there?

Reply:
That approach would work here (provided by "be able to call the original twice" means "be able to call the original an arbitrary amount of times")

Reply:
Let's apply it then, for the sake of consistency. (The other PR will hopefully be merged soon as well.)